### PR TITLE
Updated README.md

### PR DESCRIPTION
**Pull request by Soumyajit2825**
### All about Time-Complexity and Space-Complexity
@msboffl i have done the job ❣️ ....Doing a PR ..Hope you will find it good. **Thank you** ❤️

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,185 @@
...
+
+**Time complexity is defined in terms of how many times it takes to run a given algorithm, based on the length of the input. Time complexity is not a measurement of how much time it takes to execute a particular algorithm because such factors as programming language, operating system, and processing power are also considered.
+Time complexity is a type of computational complexity that describes the time required to execute an algorithm. The time complexity of an algorithm is the amount of time it takes for each statement to complete. As a result, it is highly dependent on the size of the processed data. It also aids in defining an algorithm's effectiveness and evaluating its performance.**
+
+#### Example:  
+Suppose a ❔problem is to find whether a pair (X, Y) exists in an array, A of N elements whose sum is Z. The simplest 🤔idea is to consider every pair and check if it satisfies the given condition or not.
+
+The _pseudo-code_ is as follows:
+
+int a[n];
```
Comment (msboffl) [inline README.md:13]: Can you please add the code in backticks?

Refer this links - [link](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)

File: README.md [hunk 2]
```diff
@@ -1 +1,185 @@
...
+  
+
+for(int i = 0;i < n;i++)
+  for(int j = 0;j < n;j++)
+    if(i!=j && a[i]+a[j] == z)
+       return true
+
+return false
+
+**Python Code of the above _pseudo-code_**
```
Comment (msboffl) [inline README.md:25]: Same as above comment

File: README.md [hunk 3]
```diff
@@ -1 +1,185 @@
...
+## 👾Space Complexity
+
+**When an algorithm is run on a 💻computer, it necessitates a certain amount of 📝memory space. The amount of memory used by a program to execute it is represented by its space complexity. Because a program requires memory to store input data and temporal values while 🏃running, the space complexity is auxiliary and input space.**
+
+#### Example:
+Suppose a ❓problem to 👓find the frequency of array elements.
+
+The _pseudo-code_ is as follows: 
+
+int freq[n];
```
Comment (msboffl) [inline README.md:75]: Same as above comment

File: README.md [hunk 4]
```diff
@@ -1 +1,185 @@
...
+int a[n];
+
+for(int i = 0; i<n; i++)
+{
+   cin>>a[i];
+   freq[a[i]]++;
+}  
+
+**Python Code of the above _pseudo-code_**
```
Comment (msboffl) [inline README.md:84]: Same as above comment

**Review by Soumyajit2825 — COMMENTED** [review]
@msboffl changes have been made succesfully....Check it please.Thank you

**Review by msboffl — APPROVED** [review]

**Comment by Soumyajit2825** [comment]
@msboffl this contribution can be accepted in Hacktoberfest? 😞

**Comment by msboffl** [comment]
> @msboffl this contribution can be accepted in Hacktoberfest? 😞

I have already added labels, i think accepted

**Comment by Soumyajit2825** [comment]
ok thank you 😇

**Pull request merged**